cleverhans/model.py

```diff
@@ -23,60 +32,32 @@ def __call__(self, *args, **kwargs):
         """
         return self.get_probs(*args, **kwargs)
 
-    def get_layer(self, x, layer):
-        """
-        Expose the hidden features of a model given a layer name.
-        :param x: A symbolic representation (Tensor) of the network input
-        :param layer: The name of the hidden layer to return features at.
-        :return: A symbolic representation (Tensor) of the hidden features
-        :raise: NoSuchLayerError if `layer` is not in the model.
-        """
-        # Return the symbolic representation (Tensor) for this layer.
-        output = self.fprop(x)
-        try:
-            requested = output[layer]
-        except KeyError:
-            raise NoSuchLayerError()
-        return requested
-
-    def get_logits(self, x):
+    def get_logits(self, x, **kwargs):
         """
         :param x: A symbolic representation (Tensor) of the network input
         :return: A symbolic representation (Tensor) of the output logits
         (i.e., the values fed as inputs to the softmax layer).
         """
-        return self.get_layer(x, 'logits')
+        return self.fprop(x, **kwargs)[self.O_LOGITS]
 
-    def get_probs(self, x):
+    def get_probs(self, x, **kwargs):
         """
         :param x: A symbolic representation (Tensor) of the network input
         :return: A symbolic representation (Tensor) of the output
         probabilities (i.e., the output values produced by the softmax layer).
         """
-        try:
-            return self.get_layer(x, 'probs')
-        except NoSuchLayerError:
-            pass
-        except NotImplementedError:
-            pass
-        import tensorflow as tf
-        return tf.nn.softmax(self.get_logits(x))
-
-    def get_layer_names(self):
-        """
-        :return: a list of names for the layers that can be exposed by this
-        model abstraction.
-        """
-
-        if hasattr(self, 'layer_names'):
-            return self.layer_names
+        d = self.fprop(x, **kwargs)
+        if self.O_PROBS in d:
```

Comment:
I think this behavior is potentially misleading. Is there a particular reason needed to have this return the logits if probs are not found? I would rather have it fail if probs were not found because there is a get_logits method to return the logits

Reply:
It does not return logits but softmax(logits), which generate probabilities. I don't see why that would be confusing, isn't it typical to apply a softmax to logits to obtain probabilities?

Reply:
Yeah, @david-berthelot is right, it returns the correct value